### PR TITLE
new writer

### PR DESCRIPTION
**Pull request by qubotong**
Wrting to new scene files by appending to the file instead of saving all nodes in memory.

### CODE REVIEW

File: main.py [hunk 1]
```diff
@@ -123,6 +123,7 @@ def resize(viewer):
 argparser.add_argument("-v", "--visualize", help="Visualize the kernels we are going to draw", action="store_true")
 argparser.add_argument("-b", "--bidir", help="Use bidirectional path tracing instead of path tracing for "
                                                  "incompleteness images", action="store_true")
+argparser.add_argument("-i", "--inc", help="output incompleteness scenes", action="store_true")
```
Comment (fwilliams) [inline main.py:126]: This mode is for only outputting the incompleteness images?

**Review by fwilliams — COMMENTED** [review]
Just to confirm, you've tested this commit with a real scene? If so, then I'll accept it.

File: main.py (hunk 1 above)
Comment (qubotong) [reply main.py:126]: Yes, since we don't need it all the time.

**Comment by qubotong** [comment]
I tested serveral scenes, by comparing the line numbers and loading them into mitsuba. If you have doubt about it, we can merge it later. I will do more tests.

**Pull request merged**